tests/utils/test_dataframe.py

```diff
@@ -0,0 +1,46 @@
+"""Tests for dataframe utilities."""
+
+import warnings
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from baybe.utils.dataframe import df_add_noise_to_degenerate_rows
+
+
+def test_degenerate_rows():
+    """Test noise-based deduplication of degenerate rows."""
+    # Create random dataframe
+    df = pd.DataFrame(np.random.randint(0, 100, size=(5, 3))).astype(float)
+
+    # Manually create some degenerate rows
+    df.loc[1] = df.loc[0]  # Make row 1 identical to row 0
+    df.loc[3] = df.loc[2]  # Make row 3 identical to row 2
+    df.iloc[:, -1] = 50.0  # Make last column constant to test the edge case
+
+    # Add noise
+    df_add_noise_to_degenerate_rows(df)
+
+    # Assert that the utility fixed the degenerate rows
+    assert not df.duplicated().any(), "Degenerate rows were not fixed by the utility."
+
+
+def test_degenerate_rows_invalid_input():
+    """Test that the utility correctly handles invalid input."""
+    # Create random dataframe
+    df = pd.DataFrame(np.random.randint(0, 100, size=(5, 3))).astype(float)
+
+    # Manually create some degenerate rows
+    df.loc[1] = df.loc[0]  # Make row 1 identical to row 0
+    df.loc[3] = df.loc[2]  # Make row 3 identical to row 2
+
+    # Insert invalid data types
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=FutureWarning)
```

Comment:
Can you explain what is being suppressed here?

Reply:
the lines below change entries into an otherwise numerical column to string, present pandas complains that future pandas will complain

Reply:
I see. But wouldn't then the proper approach be to change the lines such that they comply with the suggested version, instead of silencing a warning where we know that a future version will bring problems?

Reply:
its super annoying. The original error is ` FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'A' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
    df.iloc[:, -1] = "A"`

so what is the `suggested versions` ?
This also produced it (and is annoying):
```python
    df.iloc[:, -1] = df.iloc[:, -1].astype(object)
    df.iloc[:, 2] = df.iloc[:, 2].astype(object)
    df.iloc[:, -1] = "A"
    df.iloc[1, 2] = "B"
```

Reply:
Why not simply
```python
    # Insert invalid data types
    df["invalid"] = "A"
    df.loc[1, "invalid"] = "B"
```

Reply:
its not really equivalent, the second line should add a value to an existing toerhwise valid column not make a new one (for the test with 'A' thats fine i think)

Reply:
Then I don't get it – the output of my code is exactly the same as yours, provided you set the `size` in `randint` to `(5,2)` instead of `(5,3)`. Perhaps your original code does not do what you think it does? I.e. is perhaps your column selection in the "B" assignment wrong in your original code?

Reply:
true, the supposed effect is `df.iloc[1, 0] = "B"` or `df.iloc[1, 1] = "B"` and not 2

Reply:
Ok, waiting for your fix and then this PR is good to go 🚀

Reply:
sot he bug is resolved, but its unrelated to the warning you wanted fixed

ive now opted to do a very ugly `df = df.astype(object)` to avoid it
there is simply no suggestion how to else insert a single entry into a column of otherwise different dtype and I suspect it will be very annoying in pandas3

Reply:
with the merge lets wait for #364 and I'll add another test enabled by that

Reply:
alright, thx nonetheless. Feel free to resolve